File: pilhas.py
#PILHAS

pilha = []  # Cria uma pilha vazia

pilha.append(1)  # Adiciona o elemento 1 à pilha
pilha.append(2)  # Adiciona o elemento 2 à pilha        
pilha.append(3)  # Adiciona o elemento 3 à pilha

print(pilha)  # Exibe a pilha atual

topo = pilha.pop()  # Remove o elemento do topo da pilha
print("Removido:", topo)  # Exibe o elemento removido
print("Pilha após remoção:", pilha)  # Exibe a pilha após a remoção do elemento

File: pilhas.c
#include <stdio.h>

int pilha[100];
int topo = -1;

void push(int valor) {
topo++;
pilha[topo] = valor;
printf("Empilhou: %d\n", valor);
}

int pop() {
int valor = pilha[topo];
topo--;
return valor;
}

void imprimirPilha() {
    printf("Pilha [ ");
    for (int i = 0; i <= topo; i++) {
        printf("%d ", pilha[i]);
        if (i < topo) {
            printf(", ");
        }
    }
    printf("]\n");
}

int main () {
    push(1);
    push(2);
    push(3);

    imprimirPilha();

    int removido = pop();
    printf("Removido: %d\n", removido);
    imprimirPilha();

    return 0;
}